src/main/java/com/internet/shop/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.ShoppingCart;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        Storage.addCart(shoppingCart);
+        return shoppingCart;
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        List<ShoppingCart> shoppingCarts = getAll();
+        IntStream.range(0, shoppingCarts.size())
+                .filter(index
+                        -> shoppingCarts.get(index).getId().equals(shoppingCart.getId()))
+                .forEach(index -> shoppingCarts.set(index, shoppingCart));
```

Comment:
```suggestion
        IntStream.range(0, Storage.shoppingCarts.size())
                .filter(index
                        -> Storage.shoppingCarts.get(index).getId().equals(shoppingCart.getId()))
                .forEach(index -> Strorage.shoppingCarts.set(index, shoppingCart));
```
in dao you can call storage directly, no need to create separate variable